packages/dialog/src/mwc-dialog-base.ts

```diff
@@ -118,7 +120,58 @@ export class DialogBase extends BaseElement {
   }
 
   protected getInitialFocusEl(): HTMLElement|null {
-    return this.querySelector(`[${this.initialFocusAttribute}]`);
+    const initFocusSelector = `[${this.initialFocusAttribute}]`;
+
+    // only search light DOM. This typically handles all the cases
+    const lightDomQs = this.querySelector(initFocusSelector);
+
+    if (lightDomQs) {
+      return lightDomQs as HTMLElement;
+    }
+
+    // if not in light dom, search each flattened distributed node.
+    const content = this.contentSlot as HTMLSlotElement;
+    const primary = this.primarySlot as HTMLSlotElement;
+    const secondary = this.secondarySlot as HTMLSlotElement;
+
+    const contentNodes = content.assignedNodes({flatten: true});
+    const primaryNodes = primary.assignedNodes({flatten: true});
+    const secondaryNodes = secondary.assignedNodes({flatten: true});
+
+    const initFocusElement = this.searchNodeTreesForAttribute(
```

Comment:
Can you re-order these lines so that you don't run the query selector and assignNodes calls for primary and secondary when you don't need to?

Reply:
rearranged to primary -> secondary -> content since primary and secondary should have shorter trees. don't want to do work unless necessary